src/hotspot/share/runtime/thread.hpp

```diff
@@ -822,6 +822,15 @@ class Thread: public ThreadShadow {
   // Not for general synchronization use.
   static void SpinAcquire(volatile int * Lock, const char * Name);
   static void SpinRelease(volatile int * Lock);
+
+#ifdef POSIX
```

Comment:
I hate seeing this in shared code. It really belongs in platform-dependent thread code - but osThread_posix doesn't exist. :(

But do we need this - can't the existing use_vfork_if_available parameter instead be renamed and interpreted as not_in_a_signal_handler?

Reply:
But would having this functionality not make sense? We always argue about what is and is not allowed during signal handling; here, we may get a mechanism to actually assert it and e.g. prevent using malloc() or logging or whatever VM functionality may creep in.

But I will revert my last commit completely and re-add the needs-vfork parameter. This was supposed to be mainly a cleanup; maybe it was wrong to mix in behavioral changes.